core/src/main/scala/org/scalarelational/extra/PersistentProperties.scala

```diff
@@ -23,45 +25,45 @@ trait PersistentProperties extends Datastore {
 
   object persistence {
     def get(name: String): Option[String] = {
-      val query = select(persistentProperties.value) from persistentProperties where persistentProperties.key === name
+      val query = (select (persistentProperties.value)
+        from persistentProperties
+        where persistentProperties.key === name
+      )
+
       query.result.converted.headOption
     }
 
     def apply(name: String): String = {
-      get(name).getOrElse(throw new NullPointerException(s"Unable to find $name in persistent properties table."))
+      get(name).getOrElse(
+        throw new NullPointerException(s"Unable to find $name in persistent properties table."))
     }
 
-    def update(name: String, newValue: String) {
-      val m = merge(
+    def update(name: String, newValue: String): Int =
+      merge(
         persistentProperties.key,
         persistentProperties.key(name),
-        persistentProperties.value(newValue))
-      m.result
-    }
+        persistentProperties.value(newValue)).result
 
-    def remove(name: String) {
-      val d = delete(persistentProperties) where(persistentProperties.key === name)
-      d.result
-    }
+    def remove(name: String): Int =
+      (delete (persistentProperties)
+        where (persistentProperties.key === name)).result
 
-    def stringProperty(key: String, default: String = null): Property[String] = {
-      val p = Property[String](default = Some(get(key).getOrElse(default)))
-      p.change.on {
-        case evt => if (evt.newValue != null) {
-          this(key) = evt.newValue
-        } else {
-          remove(key)
-        }
+    def stringProperty(key: String): Opt[String] = {
+      val ch = new Opt[String](get(key))
+      ch.silentAttach {
```

Comment:
I don't like the naming either. Any ideas how to rename it? ``silentAttach`` does not call the handler for the initial value. It is equivalent to ``ch.tail.attach()``.